app/controllers/helper/ShapefilesCreatorHelper.java

```diff
@@ -0,0 +1,443 @@
+package controllers.helper;
+
+import java.io.*;
+import java.util.*;
+import java.util.zip.*;
+import scala.Option;
+import scala.runtime.AbstractFunction0;
+
+import org.geotools.data.DataUtilities;
+import org.geotools.data.DefaultTransaction;
+import org.geotools.data.Transaction;
+import org.geotools.data.collection.ListFeatureCollection;
+import org.geotools.data.shapefile.ShapefileDataStore;
+import org.geotools.data.shapefile.ShapefileDataStoreFactory;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.data.simple.SimpleFeatureStore;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.locationtech.jts.geom.Coordinate;
+
+import models.attribute.GlobalAttributeForAPI;
+import models.attribute.GlobalAttributeWithLabelForAPI;
+import controllers.NeighborhoodAttributeSignificance;
+import controllers.StreetAttributeSignificance;
+
+
+
+/**
+ * This example reads data for point locations and associated attributes from a comma separated text
+ * (CSV) file and exports them as a new shapefile. It illustrates how to build a feature type.
+ *
+ * <p>Note: to keep things simple in the code below the input file should not have additional spaces
+ * or tabs between fields.
+ */
+public class ShapefilesCreatorHelper {
+
+
+    public static void createGeneralShapeFile(String outputFile, SimpleFeatureType TYPE, List<SimpleFeature> features) throws Exception{
+        /*
+         * Get an output file name and create the new shapefile
+         */
+        ShapefileDataStoreFactory dataStoreFactory = new ShapefileDataStoreFactory();
+
+        Map<String, Serializable> params = new HashMap<>();
+        params.put("url", new File(outputFile + ".shp").toURI().toURL());
+        params.put("create spatial index", Boolean.TRUE);
+
+        ShapefileDataStore newDataStore =
+                (ShapefileDataStore) dataStoreFactory.createNewDataStore(params);
+
+        /*
+         * TYPE is used as a template to describe the file contents
+         */
+        newDataStore.createSchema(TYPE);
+
+        /*
+         * Write the features to the shapefile
+         */
+        Transaction transaction = new DefaultTransaction("create");
+
+        String typeName = newDataStore.getTypeNames()[0];
+        SimpleFeatureSource featureSource = newDataStore.getFeatureSource(typeName);
+        SimpleFeatureType SHAPE_TYPE = featureSource.getSchema();
+        /*
+         * The Shapefile format has a couple limitations:
```

Comment:
These comments make it look like you got this code from elsewhere. If it's a tutorial, can you add a link to it somewhere in this file? Or if it is someone else's code that deserves attribution, definitely put that in the comment at the top of the class.

Reply:
It was from the Geotools Feature tutorial. I have added a link